spec/models/planning_application_spec.rb

```diff
@@ -1068,4 +1068,24 @@
       end
     end
   end
+
+  AssessmentDetail.categories.each_key do |category|
+    describe "##{category}" do
+      let(:planning_application) do
+        create(:planning_application, :with_consultees)
+      end
+
+      let!(:assessment_detail) do
+        create(
+          :assessment_detail,
+          category,
+          planning_application: planning_application
+        )
+      end
+
+      it "returns the correct assessment detail" do
```

Comment:
It's more that I'm highlighting things I don't know about so that others, you or whoever, can be encouraged to look and correct me (or whatever).

I think my strategy would be to keep it is. If you have an error in that part of the code then you go with explicit tests along with an error case.